sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sfas-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,191 @@
+import * as path from "path";
+import { SFASIntegrationScheduler } from "../sfas-integration.scheduler";
+import { In } from "typeorm";
+import { RestrictionCode } from "@sims/services";
+
+describe(describeProcessorRootTest(QueueNames.SFASIntegration), () => {
+  let app: INestApplication;
+  let processor: SFASIntegrationScheduler;
+  let db: E2EDataSources;
+  let sftpClientMock: DeepMocked<Client>;
+  let sfasDownloadFolder: string;
+
+  beforeAll(async () => {
+    sfasDownloadFolder = path.join(__dirname, "sfas-files");
+    // Set the SFAS folder to the files mocks folders.
+    process.env.SFAS_RECEIVE_FOLDER = sfasDownloadFolder;
+    const { nestApplication, dataSource, sshClientMock } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sftpClientMock = sshClientMock;
+    // Processor under test.
+    processor = app.get(SFASIntegrationScheduler);
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    // Empty the student restrictions table for each test.
+    const studentRestrictionIds = await db.studentRestriction.find({
+      select: { id: true },
+    });
+    const ids = studentRestrictionIds.map(
+      (studentRestrictionId) => studentRestrictionId.id,
+    );
+    await db.studentRestriction.delete({ id: In(ids) });
+  });
+
+  it("Should save only one legacy restriction for the student if one or more restrictions are imported from SFAS which do not have a mapping to the SIMS restrictions", async () => {
+    const studentExists = await db.student.findOne({
+      where: {
+        birthDate: getISODateOnlyString(new Date("1998-03-24")),
+        user: { lastName: "FOUR" },
+        sinValidation: { sin: "900041310" },
+      },
+    });
+    let student = studentExists;
+    if (!studentExists) {
+      student = await saveFakeStudent(db.dataSource);
+      // Update the student to ensure that the student imported from SFAS is the same student as the one created above.
+      student.birthDate = getISODateOnlyString(new Date("1998-03-24"));
+      student.user.lastName = "FOUR";
+      student.sinValidation.sin = "900041310";
+      await db.student.save(student);
+    }
+    // Queued job.
+    const job = createMock<Job<void>>();
+    mockDownloadFiles(sftpClientMock, [SFAS_LEGACY_RESTRICTION_FILENAME]);
+    // Act
+    await processor.processSFASIntegrationFiles(job);
+    // Assert
+    // Expect the file was deleted from SFTP.
+    expect(sftpClientMock.delete).toHaveBeenCalled();
+    // Expect only one Legacy restriction to be inserted.
+    const studentRestrictionsCount = await db.studentRestriction.count({
+      where: {
+        student: { id: student.id },
+        restriction: { restrictionCode: "LGCY" },
+      },
+    });
+    expect(studentRestrictionsCount).toBe(1);
+  });
+
+  it(
+    "Should not insert a restriction to SIMS student restrictions if the same restriction is imported from SFAS but is already active in SIMS, " +
```

Comment:
A test case description that we follow has a `should` and `when` . 

When there are lot of `shoulds` then try to add one or more tests to distribute the scenarios.

Reply:
For the 2nd part of the comment, I didn't do it intentionally because this way (testing it together) helps in testing the scenarios in an integrated way rather than testing them all isolated.

Reply:
we can append the second condition with `and`.

![image](https://github.com/bcgov/SIMS/assets/54600590/01177e6b-d5f8-4e5d-8f77-d2dfe4936df8)